plugins/catalog/src/components/CatalogTable/index.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Does it also make sense to export the column functions from`columns.tsx` so people can use those functions in an explicit order rather than depending on index's in the default columns array?

Reply:
I exported the column functions from CatalogTable here: https://github.com/backstage/backstage/pull/5593/files#diff-2b7307fea1e89accb79f0f1bd27370bd5c0897566e2b3d85c76e99a841677f0eR178 so when someone wants to customize the columns, they can go to App.tsx, and define the custom columns they want like:
```
const columns = [
  CatalogTable.columns.createMetadataDescriptionColumn(),
];
...
<Route path="/catalog" element={<CatalogIndexPage columns={columns} />} />
``` 

I could export the functions directly from columns.tsx, though! Let me know if you think that is easier to understand/cleaner. Currently, I have it packaged in with CatalogTable because that is similar to [EntityTable](https://github.com/backstage/backstage/blob/master/plugins/catalog-react/src/components/EntityTable/EntityTable.tsx#L77) and I wanted to match that.

Reply:
hey folks,

I struggle to understand the behaviour of backstage here.
I need to have a custom view for the kind `Component`. When I define as above described the view like so:
```
const columns = [
  CatalogTable.columns.createMetadataDescriptionColumn(),
];
...
<Route path="/catalog" element={<CatalogIndexPage columns={columns} />} />
```
then this view is applied to all kinds.

I would like to only have the view customised for `Components`.

Where does the logic live, that tells backstage to render e.g. Domains with only the columns 
* Name
* Owner
* Description
* Tags
?

In an idea world, I'd only overwrite the view for components and leave the other views as the default set of columns.